app/main/views/suppliers.py

```diff
@@ -174,6 +174,30 @@ def approve_agreement_for_countersignature(agreement_id):
     ))
 
 
+@main.route('/suppliers/agreements/<agreement_id>/unapprove', methods=['POST'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def unapprove_agreement_for_countersignature(agreement_id):
+    # not properly validating this - all we do is pass it through
+    next_status = request.args.get("next_status")
+
+    agreement = data_api_client.unapprove_agreement_for_countersignature(
+        agreement_id,
+        current_user.email_address,
+        current_user.id,
+    )["agreement"]
+    organisation = request.form['nameOfOrganisation']
+    flash(u'The agreement for {} had its approval cancelled. You can approve it again at any time.'
```

Comment:
There is a thing in the tech debt Trello about this - I think we should consolidate all flash messages to e.g. the base page template, keep the message content there, and make the way we pass flash messages consistent.  But no, this PR isn't the place to do it.